### PR TITLE
ledger-offset: Move `Offset` to a new package. [KVL-1002]

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
Similar to #10279, this extracts `Offset` from the _participant-state_ package into its own, allowing us to further decouple _participant-state-index_ from _participant-state_.

It's a big PR, but it's about 95% imports and 2% moving a file. The only real change is the addition of the new Bazel/Maven package.

### Changelog

- **[Integration Kit]** The `Offset` type has been moved to a new Maven package, `ledger-offset`, from the `participant-state` package. The Java package has been renamed to `com.daml.ledger.offset`. If you are using this type, you will need to update your dependencies and imports.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/AdaptedV1ReadService.scala [hunk 1]
```diff
@@ -25,7 +26,7 @@ class AdaptedV1ReadService(delegate: v1.ReadService) extends ReadService {
 
   override def stateUpdates(beginAfter: Option[Offset]): Source[(Offset, Update), NotUsed] =
     delegate
-      .stateUpdates(beginAfter.map(offset => v1.Offset(offset.bytes)))
+      .stateUpdates(beginAfter.map(offset => Offset(offset.bytes)))
```
Comment (miklos-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/AdaptedV1ReadService.scala:29]:
```suggestion
      .stateUpdates(beginAfter)
```

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/AdaptedV1WriteService.scala [hunk 2]
```diff
@@ -81,7 +82,7 @@ class AdaptedV1WriteService(delegate: v1.WriteService) extends WriteService {
       submissionId: SubmissionId,
   ): CompletionStage[PruningResult] =
     delegate
-      .prune(v1.Offset(pruneUpToInclusive.bytes), submissionId)
+      .prune(Offset(pruneUpToInclusive.bytes), submissionId)
```
Comment (miklos-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/AdaptedV1WriteService.scala:85]:
```suggestion
      .prune(pruneUpToInclusive, submissionId)
```

**Review by miklos-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilder.scala [hunk 3]
```diff
@@ -3,20 +3,20 @@
...
+  * * If you have one record per block then just use [[OffsetBuilder.fromLong(<block-ID>)]]
+  * * If you may have multiple records per block then use [[OffsetBuilder.fromLong(<block-ID>, <index>)]],
+  * where <index> denotes the position or index of a given log entry in the block.
   *
-  *  @see com.daml.ledger.participant.state.v1.Offset
-  *  @see com.daml.ledger.participant.state.kvutils.api.KeyValueParticipantStateReader
+  * @see com.daml.ledger.offset.Offset
+  * @see com.daml.ledger.participant.state.kvutils.api.KeyValueParticipantStateReader
   */
 object OffsetBuilder {
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilder.scala:21]: I was wondering if this makes sense to move utilities like this to the new package. However, it belongs here probably.

**Review by hubert-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilder.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilder.scala:21]: This is particular to KV offsets. `Offset` is a generic type for all ledgers and does not follow any specific format; it just guarantees lexical ordering.

**Review by fabiotudone-da — APPROVED** [review]

**Pull request merged**